invoker/core/src/main/java/com/google/cloud/functions/invoker/runner/Invoker.java

```diff
@@ -393,6 +398,20 @@ private HttpServlet servletForDeducedSignatureType(Class<?> functionClass) {
     throw new RuntimeException(error);
   }
 
+  private ServletContextHandler addDosFilterForRequestTimeout(
+      ServletContextHandler servletContextHandler) {
+    String timeoutSeconds = System.getenv("CLOUD_RUN_TIMEOUT_SECONDS");
+    if (timeoutSeconds == null) {
+      return servletContextHandler;
+    }
+    double seconds = Double.parseDouble(timeoutSeconds);
+    long milliseconds = (long) (seconds * 1000);
+    FilterHolder holder = new FilterHolder(DoSFilter.class);
```

Comment:
The filter approach is definitely elegant, but I'm not sure we should be adding a dependency on jetty-servlets and DoSFilter just for timeout support.  I was peeking at the implementation of DoSFilter and it looks like we should be able to implement a TimeoutFilter based on just the timer/timeout stuff in https://github.com/dekellum/jetty/blob/master/jetty-servlets/src/main/java/org/eclipse/jetty/servlets/DoSFilter.java.  The interface looks pretty simple to implement, as long as we don't have to do all of the QoS/DoS stuff around suspensions, continuations, priority, rate limiting, and so on.  What do you think?